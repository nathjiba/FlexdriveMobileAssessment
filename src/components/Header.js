import React from 'react';
import {Text, View, Image, TouchableOpacity, Dimensions} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import ImageButton from './ImageButton'
import backIcon from '../assets/back_arrow.png'
import paperPlan from '../assets/paper_plane.png'
import filter from '../assets/filter_icon.png'

const {width} = Dimensions.get('window');

const Header = props => {
    return (
        <View style={styles.mainContainer}>
            <TouchableOpacity style={styles.sideComponentParent}>
                <Image style={styles.backImageStyle} source={backIcon}></Image>
            </TouchableOpacity>
            <View style={styles.headerTextParent}>
                <Text style={styles.headerTextStyle} numberOfLines={1} ellipsizeMode={'tail'}>{props.count > 0
                        ? props.count + " CARS FOUND"
                        : "NO CARS FOUND"}</Text>
                <View style={styles.subTitleParent}>
                    <Image source={paperPlan} style={styles.locationImageStyle}></Image>
                    <Text style={styles.subTitle}>{props.location}</Text>
                </View>
            </View>
            <View style={styles.sideComponentParent}>
                <ImageButton image={filter} text={'FILTER'}></ImageButton>
            </View>
        </View>
    );
}

const styles = EStyleSheet.create({
    mainContainer: {
        height: 56,
        backgroundColor: 'white',
        flexDirection: 'row',
        alignItems: 'center'
    },
    headerTextStyle: {
        fontSize: 16,
        fontWeight: '600'
    },
    // TODO : I have used hardcoded image style, when I will get proper image like
    // 1x,2x and 3x, no need to give hardcoded size
    locationImageStyle: {
        height: 8,
        width: 8
    },
    // TODO : I have used hardcoded image style, when I will get proper image like
    // 1x,2x and 3x, no need to give hardcoded size
    backImageStyle: {
        height: 28,
        width: 28
    },
    // Need to adjust filter button on small sceen size devices like iphonSE and
    // want to make text center so divided space equally
    sideComponentParent: {
        flex: width <= 320
            ? 0.63
            : 0.60,
        height: '100%',
        justifyContent: 'center'
    },
    subTitle: {
        fontSize: 12,
        fontWeight: '500'
    },
    subTitleParent: {
        flexDirection: 'row',
        alignItems: 'center',
        marginTop: 3
    },
    headerTextParent: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center'
    }

});

export default Header;
