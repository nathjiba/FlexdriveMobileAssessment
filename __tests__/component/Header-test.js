import React from 'react';
import Header from '../../src/components/Header';
import renderer from 'react-test-renderer';

test('should render without crash', () => {
    const tree = renderer
        .create(<Header count={10} location={"ATLANTA, GA"}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});