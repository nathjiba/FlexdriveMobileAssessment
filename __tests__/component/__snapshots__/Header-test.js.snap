// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crash 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/back_arrow.png",
        }
      }
    />
  </View>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
    >
      10 CARS FOUND
    </Text>
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/paper_plane.png",
          }
        }
      />
      <Text>
        ATLANTA, GA
      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/filter_icon.png",
          }
        }
      />
      <Text>
        FILTER
      </Text>
    </View>
  </View>
</View>
`;
